include/arch/arm/aarch64/lib_helpers.h

```diff
@@ -153,6 +153,21 @@ static inline bool is_el_implemented(unsigned int el)
 	return (((read_id_aa64pfr0_el1() >> shift) & ID_AA64PFR0_ELX_MASK) != 0U);
 }
 
+static inline bool is_el_highest(void)
+{
+	uint64_t el;
+	unsigned int shift;
+
+	el = GET_EL(read_currentel());
+	if (el == MODE_EL3) {
+		return true;
+	}
+
+	shift = ID_AA64PFR0_EL1_SHIFT * (el + 1);
+
+	return ((read_id_aa64pfr0_el1() >> shift) & ID_AA64PFR0_ELX_MASK) ? false : true;
```

Comment:
Not convinced this is correct.

Let's say that we enter Zephyr at `EL2` and we have something else running at `EL3`. This function is returning `true` because `EL3` is implemented but we should return `false` instead because we are running at `EL2` that is clearly not the highest implemented EL.

I think we can make this a bit easier:
1. Read  `ID_AA64PFR0_EL1`
2. Find the highest field `ELx` that is not `0x0000` starting from `EL3`
3. Compare this `EL` with the `CurrentEL`

Reply:
`shift = ID_AA64PFR0_EL1_SHIFT * (el + 1);` I have added 1 to el, so if zephyr runs in EL2, the shift will be 12 to check whether EL3 existed, if existed, return false; if not existed, return true. This issue should be zephyr runs in EL1, but we have EL3, not have EL2, there will be issue here. I'll redo this piece code.

Reply:
Updated, please check.